src/cargo/util/semver_ext.rs

```diff
@@ -182,16 +182,16 @@ impl PartialVersion {
         }
     }
 
-    pub fn exact_req(&self) -> VersionReq {
-        VersionReq {
-            comparators: vec![Comparator {
-                op: semver::Op::Exact,
-                major: self.major,
-                minor: self.minor,
-                patch: self.patch,
-                pre: self.pre.as_ref().cloned().unwrap_or_default(),
-            }],
-        }
+    pub fn matches(&self, version: &Version) -> bool {
```

Comment:
Can this method have a doc comment explaining that it matches build metadata, and that is a distinct decision that that caller needs to make whether or not that is what they want, since it is a subtle distinction?  (As compared to `Version::eq` versus `Version::cmp_precedence`, which is a subtle point that is easy to get wrong.)